design/comment-scanning.md

```diff
@@ -0,0 +1,159 @@
+/* polymer-lint enable: undefined-elements */
+```
+
+Instead of building one-off support for this into the analyzer, we’d like to also think about how the analyzer should analyze all comment-based information so that future features don’t need to reinvent the wheel when it comes to how comments should be scanned.
+
+
+## Design Overview
+
+This design groups all analysis-related comments into two supported formats: **Comment Directives** & **Comment Annotations**. **Comment Directives** are standalone features represented by standalone code comments. **Comment Annotations** are additional information/metadata that always describe the existing feature they are "attached" to.
+
+Each supports a different use-case for analysis-related documentation. Together they cover most use cases for passing instructions and metadata to the analyzer in a way that is easy for the user to extend with 3rd party support.
+
+
+### Comment Directive
+
+```
+/* polymer-lint disable */
+/* polymer-lint disable: undefined-elements, rule-2 */
+/* polymer-lint enable: undefined-elements */
+/* lazy-import: '../some-url/some-file.html' */
+<!-- polymer-build:register-service-worker-here -->
+```
+
+- A comment directive represents a single new **Directive** feature type within a document
+- A comment directive may include "arguments", which will be parsed by the scanner
+- A comment directive must exist within its own comment so that the analyzer can properly create a feature that matches that comment node's location
+- Because of the wide range of possible features these could support, specific comment syntax is lax. However they should follow these guidelines:
+  - it is unique enough to never collide with other directive formats
+  - it is formal enough to be matched by substring or RegEx matching
+  - if it takes arguments they should be parsed in the scanning phase
+- Each comment directive will have a scanner responsible for scanning for it.
+  - A scanner may be responsible for scanning multiple directives (ex: `polymer-lint enable`, `polymer-lint disable`)
+Here is a definition for the **Directive** class:
+
+```
+class Directive extends Feature {
+  identifier: string;
+  args?: string[];
+}
+```
+
+And here are how those example comment directive's above would be parsed:
+
+```
+/* polymer-lint disable */
+Directive({identifier: 'polymer-lint disable', args: null})
+/* polymer-lint disable: undefined-elements, rule-2 */
+Directive({identifier: 'polymer-lint disable', args: ['undefined-elements', 'rule-2']})
+/* polymer-lint enable: undefined-elements */
+Directive({identifier: 'polymer-lint enable', args: ['undefined-elements']})
+/* lazy-import: '../some-url/some-file.html' */
+Directive({identifier: 'lazy-import', args: ['../some-url/some-file.html']})
+<!-- polymer-build:register-service-worker-here -->
+Directive({identifier: 'polymer-build:register-service-worker-here', args: null})
+```
+
+
+### Comment Annotation
```

Comment:
I would just kill this section. we already handle this and it doesn't seem tied into the the other directives in any way.